fmralign/pairwise_alignment.py

```diff
@@ -0,0 +1,310 @@
+        If True, clustering of voxels is performed jointly using X_ and Y_ signal. If false, it's performed only on X_
+    clustering_index : list of integers
+        Clustering is performed on a 20% subset of the data chosen randomly in timeframes. This index carry this subset.
+
+    Returns
+    -------
+    alignment_algo
+        Instance of alignment estimator class fitted for X_i, Y_i
+    """
+    if n_pieces > 1:
+        if joint_clustering:
+            clustering_data = np.hstack(
+                X_[:, clustering_index], Y_[:, clustering_index])
+        else:
+            clustering_data = X_[:, clustering_index]
+        labels = make_parcellation(clustering_data, mask,
+                                   n_pieces, clustering_method, memory=mem)
+    else:
+        labels = np.zeros(int(mask.sum()), dtype=np.int8)
+
+    fit = Parallel(n_jobs, backend="threading", verbose=verbose)(
+        delayed(fit_one_piece)(
+            X_i, Y_i, alignment_method
+        ) for X_i, Y_i in generate_Xi_Yi(labels, X_, Y_)
+    )
+
+    return labels, fit
+
+
+class PairwiseAlignment(BaseEstimator, TransformerMixin):
+    """
+    Decompose the source and target images into source and target regions
+     Use alignment algorithms to align source and target regions independantly.
+    """
+
+    def __init__(self, alignment_method, n_pieces=1, clustering_method='k_means', joint_clustering=False, n_bags=1, perturbation=False, mask=None, smoothing_fwhm=None, standardize=None, detrend=False, target_affine=None, target_shape=None, low_pass=None, high_pass=None, t_r=None, memory=Memory(cachedir=None), memory_level=0, n_jobs=1, verbose=0):
+        """
+        Use alignment algorithms to align source and target images.
+        If n_pieces > 1, decomposes the images into regions and align each source/target region independantly.
+        If n_bags > 1, this parcellation process is applied multiple time and the resulting models are bagged.
+
+        Parameters
+        ----------
+        alignment_method: string
+        algorithm used to perform alignment between regions
+        'identity', 'scaled_orthogonal', 'ridge_cv', 'permutation', 'optimal_transport' or an instance of one of alignment classes provided in functional_alignment.alignment_methods
+        n_pieces: int, optional
+            number of regions in which the data is parcellated for alignment
+            if 1 the alignment is done on full scale data, if >1, the voxels are clustered and alignment is performed on each cluster applied to X and Y.
+        clustering_method : string, optional
+            'ward' or 'k_means', method used for clustering of voxels
+        joint_clustering : bool, optional
+            if True, the clustering is done jointly on X and Y, if False it is done only using X signal
+        n_bags: int, optional
+            For parcellation alignments, number of bagged parcellations used.
+        perturbation: bool, optional
```

Comment:
defaults to False

Reply:
I don't usually repeat which value of a keyword argument is the default one in docstring as they are written just above in the signature. Should I?